frontend/src/container/SearchFilterMenu.jsx

```diff
@@ -63,6 +71,45 @@ export class SearchFilterMenu extends React.Component {
     }))
   }
 
+  handleChangeNewestAuthoredContentDate = (date, type) => {
+    const { props } = this
+    if (type === DATE_FILTER_ELEMENT.AFTER) {
+      this.setState(prev => ({
+        newestAuthoredContentRange: {
+          ...prev.newestAuthoredContentRange,
+          afterDateActive: true,
+          afterDate: date
+        }
+      }))
+      props.onChangeNewestAuthoredContentDate({ from: date })
+    } else {
+      this.setState(prev => ({
+        newestAuthoredContentRange: {
+          ...prev.newestAuthoredContentRange,
+          beforeDateActive: true,
+          beforeDate: date
+        }
+      }))
+      props.onChangeNewestAuthoredContentDate({ to: date })
+    }
+  }
```

Comment:
I think this handler doesn't need to switch between the two values, instead write two handlers (one for after, the other for before) and use the right one when creating the component.